### ISSUE TITLE
RetroAchievements support

### ISSUE THREAD

**Issue by eVenent** [schellingb/dosbox-pure#362]
Recently standalone PCSX2 emulator receive support of [retroachievements](http://retroachievements.org/) and it works great. I believe there would be a big interest in classic DOS games achievements too. Is it possible to make Dosbox Pure compatible with RALib Support? I've checked their [official documentation](https://docs.retroachievements.org/Emulator-Support-and-Issues/) and DOSbox Pure has status "needs work" with these points:
• Needs a reliable hashing method
• Needs a way to prevent launching with user-specified command-line parameters
• Needs a way to prevent TSRs

Is it possible to fix it and enable achievements for DOS games?

**Comment by Darknior** [comment]
Woaw crazzy, for sure it will be a great feature to add !!!
RA are played by so many gamers, and make them discover so many games ... i love the idea :)

**Comment by yonailo** [comment]
yes, supporting !!

**Comment by schellingb** [comment]
Hopefully soon. But I would have said that a year ago :-) Technical issues that have been thought of so far have been solved both on the core side and the RetroAchievement side over the last 2 years. New technical problems might still crop up once the team over at RetroAchievements and achievement developers actually start evaluating the platform. As far as I know they are adding GameCube support currently. We'll see what will happen once that is all wrapped up.

**Comment by bonswouar** [comment]
Do you know if they've found a solution regarding anti cheat?
As DOS is a full OS you can install and tweak pretty much anything, which might be a problem as far as I understand

**Comment by dancek** [comment]
Is the [list of blockers in RA docs](https://docs.retroachievements.org/developer-docs/unsupported-emulators-and-cores.html#ms-dos) up to date?

> ❌ libretro core: dosbox-pure
> - Needs a way to prevent launching with user-specified command line parameters
> - Needs a way to prevent TSRs
> - Needs a way to deactivate achievements if game drops to command prompt

I'd be happy to help and could try working on some of these. I've read some dosbox code in the past but haven't contributed.

**Comment by schellingb** [comment]
No, it seems to be not up to date. As I mentioned last year, we have solved the "hardcore mode" requirements. Some people of the RetroAchievement community are working on documenting known game releases so they can be fixed via the hashing method I contributed to their project at the end of 2023. So the technological pieces are in place, but long-term logistical tasks remain. I don't really know when things can go into the next phase though. Not every DOS game release needs to be documented and hashed, but a good amount of "interesting" games is my guess. I think some progress is being reported on the RetroAchievements Discord instance.

**Comment by dancek** [comment]
Thanks! I'll check their discord to see if there's anything technical that needs help!

**Comment by WedgeSparda** [comment]
Hi, any update related to this? 🙏
